### DISCUSSION TITLE
ONVIF testing needed?

### DISCUSSION THREAD

**Discussion by flarfnug** [glikely/obs-ptz#137] [General]
I would love to use this plugin, but I have an Onvif camera.  What do you need to get it working? I bet this plugin would suddenly become WAY more popular if it were enabled.

**Comment by glikely** [comment]
I had someone contribute ONVIF support, but unfortunately the code as written is unsafe and will block the OBS UI thread. Also, the contributor has since disappeared and hasn't addressed any of the problems.

What I really need is someone to fix the code so that it isn't risky to enable it.

**Reply by flarfnug** [reply]
I figured out how to do what I needed to do with the least amount of programming.  This is for a Reolink E1 Outdoor camera and will move the camera to the preset at id #.  Replace the IP address and #user# and #password# with your own.  I'm not worried about sending a plaintext user and password over my LAN since it's all firewalled. 
`curl -s -k -X POST -H "Content-Type : application/json" -d "[{\"cmd\":\"PtzCtrl\",\"action\":0,\"param\":{\"channel\":0,\"id\":1,\"op\":\"ToPos\",\"speed\":32}}]" "https://192.168.254.xxx/cgi-bin/api.cgi?user=#user#&password=#password#"`
This is from: [https://www.reddit.com/r/reolinkcam/comments/wq371h/comment/ikn1zex/?utm_source=share&utm_medium=web2x&context=3](url)  User: [https://www.reddit.com/user/mnpg_fr/](url) figured it out.

**Comment by flarfnug** [comment]
The experimental ONVIF controls work fine.  Why do you have to put down ONVIF camera's so hard? I'm a hobbyist Youtuber, not a Television station.  Lol!

**Reply by glikely** [reply]
> The experimental ONVIF controls work fine.

Glad it’s working for you!

> Why do you have to put down ONVIF camera's so hard? I'm a hobbyist Youtuber, not a Television station. Lol!

Because it is true! All the CCTV cameras I’ve looked at have poor frame rate and poor latency. For anyone in the market for a camera, I would recommend a second hand VISCA camera off eBay before a CCTV camera. I’m using second hand Sony VISCA cameras myself.

But that’s just my advice to anyone buying a camera now. I still want to have really good support for ONVIF in the plugin

**Comment by glikely** [comment]
I've just tagged a second ONVIF support release. Please give it a try. Presets should work better now.

https://github.com/glikely/obs-ptz/releases/tag/v0.16.0-test2

**Reply by vicesalles-uab** [reply]
when adding the presents by the "+" button, a new item appears on the list but it doesn't work. By using the "add preset" on the contextual menu of the presets list, no item gets added on the list but when restarting obs a working preset appears on the list. Beyond that, no crashes or UI freezing noticed. It works smoothly.

I'm using Minrray cameras.

**Comment by athayde2** [comment]
excellent, I used a ptz hikvision camera with onvif and everything worked, congratulations

**Reply by stixoffire** [reply]
I do not have a camera in the drop down list - what do I need to do to add it in ?
![2024_12_25_OBS_Window](https://github.com/user-attachments/assets/ed184fd5-4d1a-45f2-a9cb-fdd44af57066)

**Comment by stixoffire** [comment]
I am trying to ad din the device on OBS Version 31 , in Windows 10 . I do not see how I can add IP url or anything to get IP Camera in . 
Does anyone know how I can do this?
![2024_12_25_OBS_Window](https://github.com/user-attachments/assets/c5633f11-efea-4313-9a80-58f45c418bf1)
